prog/weaver/main.go

```diff
@@ -407,7 +407,7 @@ func main() {
 	}
 
 	if enablePlugin || enablePluginV2 {
-		go plugin.Start(httpAddr, dockerCli, pluginSocket, pluginMeshSocket, noDNS, enablePluginV2)
+		go plugin.Start(httpAddr, dockerCli, pluginSocket, pluginMeshSocket, !noDNS, enablePluginV2)
```

Comment:
Actually, this is valid.
It matches the CLI flag `-no-dns`.
It is then negated since the rest of the code cares about `dns`.